src/main/java/net/consensys/wittgenstein/protocols/ETHPoW.java

```diff
@@ -231,11 +247,15 @@ private void mine1ms() {
     @Override
     public boolean onBlock(POWBlock b) {
       boolean res = super.onBlock(b);
+
       if (res) {
         // Someone sent us a new block, so we're going to switch
         //  our mining to this new head
         inMining = null;
       }
+      if (this.blocksReceivedByBlockId.put(b.id, b) != null) {
```

Comment:
not sure about the logic here, but it seems better to look at the possible uncles when we create the new block only